ckanext/xloader/loader.py

```diff
@@ -56,6 +56,8 @@ def load_csv(csv_filepath, resource_id, mimetype='text/csv', logger=None):
             #                                          extension=format)
             # except Exception:
                 raise LoaderError('Messytables error: {}'.format(e))
+        except Exception:
+            raise FileCouldNotBeLoadedError()
```

Comment:
I don't quite get this, I'm afraid - probably needs a comment to explain. If it raised LoaderError here then the details of the error would be logged for the user and it would try loading again with messytables, including 'trying the format' (Or maybe you've checked and it doesn't in this case?). Raising FileCouldNotBeLoadedError quits with: "Loading file raised an error: FileCouldNotBeLoadedError", doesn't it?

Reply:
Yeah, this part is still not making me happy. The issue is, that at this point, we could come along with an excerpt of a a xls-file for example as we can't really rely on the `resource.get('format')` as it can be empty or even worse - contains wrong information altogether, which will then result in a corrupt file that can be processed. 
I wanted to prevent a second try in that case as the logging would mislead the user to what really went wrong (we would just end with the statement 'No entries found - nothing to load')
That is why I am now tempted to drop `EXCERPT_FORMATS` to begin with and deal with the error that will be thrown in the case of a corrupt file. Not sure if the error 'array index out of range' is a guarantee for the error that the excerpt produced though :) My next push hopefully makes this a little bit clearer.